src/udev/udev-builtin-net_id.c

```diff
@@ -641,7 +689,11 @@ static int names_pci_slot(sd_device *dev, sd_device *pci_dev, const char *prefix
                          strna(domain), bus_and_slot, strna(func), strna(port),
                          special_glyph(SPECIAL_GLYPH_ARROW_RIGHT), empty_to_na(str));
 
-        r = pci_get_hotplug_slot(pci_dev, &hotplug_slot);
+        if (naming_scheme_has(NAMING_FIRMWARE_NODE_SUN))
+                r = pci_get_slot_from_firmware_node_sun(pci_dev, &slot);
```

Comment:
if the "firmware_node/sun" file doesn't exist, shouldn't we fall back to the old ways?

Reply:
I'm now falling back to hotplug_slot so we are sure we don't break s390.
I'm not a fan of the old way, but I also don't know all the weird way a PCI tree can look, so I guess this is the safest way.